tests/test_server.py

```diff
@@ -1266,7 +1266,25 @@ async def test_resource_deploy_done(server, client, environment, agent, caplog,
     assert resource_action["action"] == const.ResourceAction.deploy
     assert resource_action["started"] is not None
     assert resource_action["finished"] is not None
-    assert resource_action["messages"] == [
+
+    def check_messages(actual_result, expected_result):
```

Comment:
If you do end up using `time-machine`, perhaps you can also get rid of this comparison approach by using the same timezone in the expected list.